graql/reasoner/ResolutionIterator.java

```diff
@@ -115,19 +132,22 @@ public boolean hasNext() {
         if (reiterate()) {
             long dAns = answers.size() - oldAns;
             if (dAns != 0 || iter == 0) {
-                LOG.debug("iter: {} answers: {} dAns = {}", iter, answers.size(), dAns);
+                LOG.debug("iter: {} answers: {} dAns = {} time = {}", iter, answers.size(), dAns, System.currentTimeMillis() - startTime);
                 iter++;
                 states.push(query.resolutionState(new ConceptMap(), new UnifierImpl(), null, new HashSet<>()));
+
                 oldAns = answers.size();
+                startTime = System.currentTimeMillis();
                 return hasNext();
             }
         }
 
         MultilevelSemanticCache queryCache = CacheCasting.queryCacheCast(this.queryCache);
-
         subGoals.forEach(queryCache::ackCompleteness);
         queryCache.propagateAnswers();
 
+        if (LOG.isTraceEnabled()) logTree.outputToFile(Paths.get("/tmp/query.profile"));
```

Comment:
I wonder if we can find a better place to put this file, would break on windows atm right

Reply:
ideally it should be in the log dir. Not sure how to access it.

Reply:
ask @vmax

Reply:
changed